core/runtime/ops.rs

```diff
@@ -173,16 +174,22 @@ pub fn map_async_op_infallible<R: 'static>(
   // of ops. If we can figure out a way around this, we can remove this call to boxed_local and save a malloc per future.
   let mut pinned = op.map(move |res| (promise_id, id, res)).boxed_local();
 
-  match pinned.poll_unpin(&mut Context::from_waker(noop_waker_ref())) {
-    Poll::Pending => {}
-    Poll::Ready(res) => {
-      if ctx.metrics_enabled() {
-        dispatch_metrics_async(ctx, OpMetricsEvent::Completed);
+  let pinned =
+    match pinned.poll_unpin(&mut Context::from_waker(noop_waker_ref())) {
+      Poll::Pending => pinned,
+      Poll::Ready(res) => {
+        // TODO(mmastrac): optimize this so we don't double-allocate
+        if deferred {
+          ready(res).boxed_local()
```

Comment:
Are deferred ops tracking metrics?

Reply:
They work but I added a test to confirm because it's not obvious.